fairing/training/native/deployment.py

```diff
@@ -0,0 +1,100 @@
+from __future__ import unicode_literals
+from __future__ import print_function
+from __future__ import division
+from __future__ import absolute_import
+from future import standard_library
+standard_library.install_aliases()
+import logging
+
+from kubernetes import client as k8s_client
+
+from fairing import config
+from fairing import utils
+from fairing.training import base
+from fairing.backend import kubernetes
+
+logger = logging.getLogger(__name__)
+DEFAULT_JOB_NAME = 'fairing-job'
+
+class NativeDeployment(object):
+    """Handle all the template building for metaparticle api and calling mpclient
+
+    Attributes:
+        namespace: k8s namespace where the training's components 
+            will be deployed.
+        runs: Number of training(s) to be deployed. Hyperparameter search
+            will generate multiple jobs.
+    """
+
+    def __init__(self, namespace, runs):
+        if namespace is None:
+            self.namespace = utils.get_default_target_namespace()
+        else:
+            self.namespace = namespace
+        
+        # Used as pod and job name
+        self.name = "{}-{}".format(DEFAULT_JOB_NAME, utils.get_unique_tag())
```

Comment:
v1.objectmeta has a nice field called "generateName" that will do this for you. That might be slightly simpler.